Sources/Security/Signing.swift

```diff
@@ -185,14 +201,74 @@ extension Signing.SignatureParameters {
 
     var asData: Data {
         return (
-            self.nonce +
+            (self.nonce ?? .init()) +
             String(self.requestDate).asData +
-            self.message
+            self.message.sha1
         )
     }
 
 }
 
 // MARK: - Private
 
+extension Signing {
+
+    enum SignatureComponent: CaseIterable, Comparable {
+
+        case intermediatePublicKey
+        case expiration
+        case intermediatePublicKeySignature
+        case salt
+        case payload
+
+        var size: Int {
+            switch self {
+            case .intermediatePublicKey: return 32
+            case .expiration: return 4
```

Comment:
Should we rename this to `intermediatePublicKeyExpiration`?

Reply:
Oh yeah